Mathlib/ModelTheory/Fraisse.lean

```diff
@@ -302,6 +365,56 @@ theorem isFraisse [Countable (Σ l, L.Functions l)] [Countable M] (h : IsFraisse
     IsFraisse K :=
   (congr rfl h.age).mp h.ultrahomogeneous.age_isFraisse
 
+variable {K} {N : Type w} [L.Structure N]
+variable [Countable (Σ l, L.Functions l)] [Countable M] [Countable N]
+variable (hM : IsFraisseLimit K M) (hN : IsFraisseLimit K N)
+
+include hM hN
+
+protected theorem IsExtensionPair : L.IsExtensionPair M N := by
+  intro ⟨f, f_FG⟩ m
+  let S := f.dom ⊔ closure L {m}
+  have dom_le_S : f.dom ≤ S := by
+    simp only [S, closure_union, closure_eq, ge_iff_le, le_sup_left]
+  have S_FG : S.FG := f_FG.sup (Substructure.fg_closure_singleton _)
+  have S_in_age_N : ⟨S, inferInstance⟩ ∈ L.age N := by
+    rw [hN.age, ← hM.age]
+    exact ⟨(fg_iff_structure_fg S).1 S_FG, ⟨subtype _⟩⟩
+  let nonempty_S_N : Nonempty (S ↪[L] N) := by
+    let ⟨_, this⟩ := S_in_age_N
+    exact this
+  let ⟨g, eq⟩ := hN.ultrahomogeneous.extend_embedding (f.dom.fg_iff_structure_fg.1 f_FG)
+    ((subtype f.cod).comp f.toEquiv.toEmbedding) (inclusion dom_le_S)
+  refine ⟨⟨⟨S, g.toHom.range, g.equivRange⟩, S_FG⟩,
+    subset_closure.trans (le_sup_right : _ ≤ S) (mem_singleton m), ?_⟩
+  simp only [Subtype.mk_le_mk, PartialEquiv.le_def]
+  use dom_le_S
+  rw [eq]
+  rfl
+
+theorem unique_FraisseLimit : Nonempty (M ≃[L] N) := by
```

Comment:
```suggestion
theorem nonempty_FraisseLimit : Nonempty (M ≃[L] N) := by
```
also what is the `FraisseLimit` referring to?

Reply:
I've added a docstring for this important theorem (although it also was in the module doc), and I'm changing the name.
It's already in the `IsFraisseLimit` namespace, so `IsFraisseLimit.nonempty_equiv` now proves that any two Fraïssé limits (`M` and `N`) admit a (not necessarily unique) isomorphism.

Reply:
Oh yeah, that makes more sense. Sorry for being slow